internal/service/alarms/internal/db/repo/alarms_repository.go

```diff
@@ -51,3 +59,193 @@ func (ar *AlarmsRepository) GetAlarmEventRecordWithUuid(ctx context.Context, uui
 	slog.Info("AlarmEventRecord found", "uuid", uuid)
 	return []models.AlarmEventRecord{record}, nil
 }
+
+// UpsertAlarmEventRecord insert and updating an AlarmEventRecord
+func (ar *AlarmsRepository) UpsertAlarmEventRecord(ctx context.Context, records []models.AlarmEventRecord) error {
+	return ar.upsertEventRecords(ctx, records, models.AlarmEventRecordView.Name(ctx).String())
+}
+
+func (ar *AlarmsRepository) upsertEventRecords(ctx context.Context, records []models.AlarmEventRecord, tableName string) error {
+	// Build queries for each record in the batch
+	sql, params, err := buildAlarmEventRecordUpsertQuery(records, tableName)
+	if err != nil {
+		return fmt.Errorf("failed to build query for event upsert: %w", err)
+	}
+
+	_, err = ar.Db.Exec(ctx, sql, params...)
+	if err != nil {
+		return fmt.Errorf("failed to execute upsert query: %w", err)
+	}
+
+	slog.Info("Successfully inserted alerts from alertmanager", "table", tableName, "count", len(records))
+	return nil
+}
+
+// buildAlarmEventRecordUpsertQuery builds the query for insert and updating an AlarmEventRecord
+func buildAlarmEventRecordUpsertQuery(records []models.AlarmEventRecord, tableName string) (string, []any, error) {
+	query := psql.Insert(im.Into(tableName))
+
+	// Dynamically construct the column list
+	columns := []string{
+		"alarm_raised_time", "alarm_changed_time", "alarm_cleared_time", "alarm_acknowledged_time",
+		"alarm_acknowledged", "perceived_severity", "extensions", "resource_id", "resource_type_id",
+		"notification_event_type", "alarm_status", "fingerprint", "alarm_definition_id", "probable_cause_id",
+	}
+
+	// Check if alarm_event_record_id is needed
+	hasAlarmEventRecordID := false
+	for _, record := range records {
+		if record.AlarmEventRecordID != nil {
+			hasAlarmEventRecordID = true
+			columns = append([]string{"alarm_event_record_id"}, columns...)
+			break
+		}
+	}
+
+	// Set the column list for the InsertQuery
+	query.Expression.Columns = columns
+
+	// Add the values for each record
+	values := make([]bob.Mod[*dialect.InsertQuery], 0, len(records))
+	for _, record := range records {
+		// Build the values for the row
+		args := []any{
+			record.AlarmRaisedTime, record.AlarmChangedTime, record.AlarmClearedTime,
```

Comment:
It's going to be tricky 👀 to inspect this to make sure the column names and the column data match, could you align the line breaks so that the same column names and data fields are on the same lines.  Maybe 3 per line each.

Reply:
Good idea! Wonder if there's a CI check we can use to catch this. But will follow 3 per line for all.